src/createServer.js

```diff
@@ -1,3 +1,44 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const { convertToCase } = require('./convertToCase/convertToCase');
+const http = require('http');
+
+const SUPPORTED_CASES = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const { pathname, searchParams } = new URL(req.url, `http://${req.headers.host}`);
+    const originalText = pathname.slice(1);
+    const targetCase = searchParams.get('toCase');
+
+    const errors = [];
+
+    if (!originalText) {
+      errors.push({ message: 'Text to convert is required. '
+        + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".' });
+    }
+
+    if (!targetCase) {
+      errors.push({ message: '"toCase" query param is required. '
+        + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".' });
+    } else if (!SUPPORTED_CASES.includes(targetCase)) {
+      errors.push({ message: 'This case is not supported. '
+        + 'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.' });
+    }
+
+    if (errors.length) {
+      res.statusCode = 400;
+      res.setHeader('Content-Type', 'application/json');
```

Comment:
You can set this header at the top of this function, as you need to always return json